dotCMS/src/main/java/com/dotcms/util/DbExporterUtil.java

```diff
@@ -0,0 +1,237 @@
+    }
+
+    /**
+     * Resolves the path for pg_dump binary.
+     * It tries to find it by calling the `which pg_dump` command. If it cannot locate the command there, then copies it
+     * from the classpath to the dynamic content path.
+     */
+    static final Lazy<String> pgDumpPath = Lazy.of(() -> {
+        final Optional<String> pgDumpOpt = findExecutable("pg_dump");
+        if (pgDumpOpt.isPresent()) {
+            return pgDumpOpt.get();
+        }
+
+        final File pgDump = new File(ConfigUtils.getDynamicContentPath() + separator + "bin" + separator + "pg_dump");
+        if (!pgDump.exists()) {
+            copyPgDump(pgDump);
+        }
+
+        return pgDump.getAbsolutePath();
+    });
+
+    /**
+     * Copies pg_dumb binary to a defined folder for it to called later.
+     *
+     * @param pgDumpFile pg_dump binary file
+     */
+    @VisibleForTesting
+    static void copyPgDump(final File pgDumpFile) {
+        pgDumpFile.mkdirs();
+        pgDumpFile.delete();
+        
+        Logger.info(DbExporterUtil.class, "Copying pg_dump from: " + PG_ARCH.pgDump() + " to " + pgDumpFile);
+        try (final InputStream in = DbExporterUtil.class.getResourceAsStream("/" + PG_ARCH.pgDump());
+             final OutputStream out = Files.newOutputStream(pgDumpFile.toPath())) {
+            System.out.println("in: " + in);
+            System.out.println("out:" + out);
+            
+            IOUtils.copy(in, out);
+        } catch (Exception e) {
+            e.printStackTrace();
+            pgDumpFile.delete();
+            throw new DotRuntimeException(e);
+        }
+        pgDumpFile.setExecutable(true);
+    }
+
+    /**
+     * Calls the pg_dumb binary with necessary arguments and redirects standard output to {@link InputStream}.
+     *
+     * @return input stream with commands output
+     * @throws IOException
+     */
+    static InputStream exportSql() throws IOException {
+        final ProcessBuilder pb = new ProcessBuilder(
+                pgDumpPath.get(),
+                getDatasource().getDbUrl(),
+                "--no-comments",
+                "-T clickstream*",
+                "-T dashboard*",
+                "-Z7",
+                "-O",
+                "-x")
+                .redirectErrorStream(true);
+        return new BufferedInputStream(pb.start().getInputStream());
+    }
+
+    /**
+     * Looks for pg_dump binary by calling the `which <binary to find>` command.
+     *
+     * @param programToFind binary to find
+     * @return path to the located binary
+     */
+    @VisibleForTesting
+    static Optional<String> findExecutable(final String programToFind) {
+        final List<String> commands = ImmutableList.<String>builder()
+                .add("which")
+                .add(programToFind)
+                .build();
+
+        Process processHolder = null;
+        try {
+            final ProcessBuilder pb = new ProcessBuilder(commands);
+            pb.redirectErrorStream(true);
```

Comment:
add some log here for support, pour guys debugging this :)

Reply:
Logged.